nano/node/rpc.cpp

```diff
@@ -244,6 +278,29 @@ nano::amount nano::rpc_handler::amount_impl ()
 	return result;
 }
 
+std::shared_ptr<nano::block> nano::rpc_handler::block_impl (bool signature_work_required)
+{
+	std::shared_ptr<nano::block> result;
+	if (!ec)
+	{
+		std::string block_text (request.get<std::string> ("block"));
+		boost::property_tree::ptree block_l;
+		std::stringstream block_stream (block_text);
+		boost::property_tree::read_json (block_stream, block_l);
+		if (!signature_work_required)
+		{
+			block_l.put ("signature", "0");
+			block_l.put ("work", "0");
```

Comment:
For atleast `sign()` it might be helpful not overwrite `work` if it was supplied.

Reply:
Nevermind, this only succeeds if the attribute doesn't already exist it seems.